### ISSUE TITLE
@payload/richtext-lexical, UploadFeature w/ fields cannot Publish before opening and saving in Edit modal

### ISSUE THREAD

**Issue by galcsikgyozo** [payloadcms/payload#6868]
### Payload Version

3.0.0-beta.53

### Node Version

20.9.0

### Next.js Version

15.0.0-rc.0

### Describe the Bug

When extending the UploadFeature with some custom fields – in this example a simple text field for filling in the alt tag optionally -, doesn't matter if the fields are required or not, initially clicking on Publish Changes produces a 500 error, and says that the field cannot be null, in the terminal.
It only accepts me saving the page, if I click on the "Edit Relationship" button, and click on "Save Changes" in the opened modal.

https://github.com/payloadcms/payload/assets/60744669/690e049b-ccd5-410a-864c-3a2950b40ac6

```
10:48:39] ERROR: TypeError: Cannot read properties of null (reading 'alt')
    at promise (webpack-internal:///(rsc)/./node_modules/.pnpm/payload@3.0.0-beta.53_@swc+core@1.6.3_@swc+types@0.1.8_graphql@16.8.2_typescript@5.4.5/node_modules/payload/dist/fields/hooks/beforeChange/promise.js:117:38)
    at eval (webpack-internal:///(rsc)/./node_modules/.pnpm/payload@3.0.0-beta.53_@swc+core@1.6.3_@swc+types@0.1.8_graphql@16.8.2_typescript@5.4.5/node_modules/payload/dist/fields/hooks/beforeChange/traverseFields.js:17:75)
    at Array.forEach (<anonymous>)
    at traverseFields (webpack-internal:///(rsc)/./node_modules/.pnpm/payload@3.0.0-beta.53_@swc+core@1.6.3_@swc+types@0.1.8_graphql@16.8.2_typescript@5.4.5/node_modules/payload/dist/fields/hooks/beforeChange/traverseFields.js:16:12)
    at hooks.beforeChange (webpack-internal:///(rsc)/./node_modules/.pnpm/@payloadcms+richtext-lexical@3.0.0-beta.53_@faceless-ui+modal@3.0.0-beta.0_@faceless-ui+scrol_cgj3cqswktc4zfgl3pxfp75cku/node_modules/@payloadcms/richtext-lexical/dist/index.js:537:99)
    at eval (webpack-internal:///(rsc)/./node_modules/.pnpm/payload@3.0.0-beta.53_@swc+core@1.6.3_@swc+types@0.1.8_graphql@16.8.2_typescript@5.4.5/node_modules/payload/dist/fields/hooks/beforeChange/promise.js:377:51)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async promise (webpack-internal:///(rsc)/./node_modules/.pnpm/payload@3.0.0-beta.53_@swc+core@1.6.3_@swc+types@0.1.8_graphql@16.8.2_typescript@5.4.5/node_modules/payload/dist/fields/hooks/beforeChange/promise.js:375:21)
    at async Promise.all (index 3)
    at async traverseFields (webpack-internal:///(rsc)/./node_modules/.pnpm/payload@3.0.0-beta.53_@swc+core@1.6.3_@swc+types@0.1.8_graphql@16.8.2_typescript@5.4.5/node_modules/payload/dist/fields/hooks/beforeChange/traverseFields.js:39:5)
    at async beforeChange (webpack-internal:///(rsc)/./node_modules/.pnpm/payload@3.0.0-beta.53_@swc+core@1.6.3_@swc+types@0.1.8_graphql@16.8.2_typescript@5.4.5/node_modules/payload/dist/fields/hooks/beforeChange/index.js:23:5)
    at async updateByIDOperation (webpack-internal:///(rsc)/./node_modules/.pnpm/payload@3.0.0-beta.53_@swc+core@1.6.3_@swc+types@0.1.8_graphql@16.8.2_typescript@5.4.5/node_modules/payload/dist/collections/operations/updateByID.js:191:22)
    at async Object.updateByID (webpack-internal:///(rsc)/./node_modules/.pnpm/@payloadcms+next@3.0.0-beta.53_graphql@16.8.2_monaco-editor@0.49.0_next@15.0.0-rc.0_payload@3_5kyp6ik2iot2zhiqfxe7yiueyu/node_modules/@payloadcms/next/dist/routes/rest/collections/updateByID.js:25:17)
    at async eval (webpack-internal:///(rsc)/./node_modules/.pnpm/@payloadcms+next@3.0.0-beta.53_graphql@16.8.2_monaco-editor@0.49.0_next@15.0.0-rc.0_payload@3_5kyp6ik2iot2zhiqfxe7yiueyu/node_modules/@payloadcms/next/dist/routes/rest/index.js:705:35)
    at async /Users/gyozo/react/m15/node_modules/.pnpm/next@15.0.0-rc.0_react-dom@19.0.0-rc-fb9a90fa48-20240614_react@19.0.0-rc-fb9a90fa48-20240614_sass@1.77.6/node_modules/next/dist/compiled/next-server/app-route.runtime.dev.js:6:63352
    at async eP.execute (/Users/gyozo/react/m15/node_modules/.pnpm/next@15.0.0-rc.0_react-dom@19.0.0-rc-fb9a90fa48-20240614_react@19.0.0-rc-fb9a90fa48-20240614_sass@1.77.6/node_modules/next/dist/compiled/next-server/app-route.runtime.dev.js:6:54549)
    at async eP.handle (/Users/gyozo/react/m15/node_modules/.pnpm/next@15.0.0-rc.0_react-dom@19.0.0-rc-fb9a90fa48-20240614_react@19.0.0-rc-fb9a90fa48-20240614_sass@1.77.6/node_modules/next/dist/compiled/next-server/app-route.runtime.dev.js:6:64693)
    at async doRender (/Users/gyozo/react/m15/node_modules/.pnpm/next@15.0.0-rc.0_react-dom@19.0.0-rc-fb9a90fa48-20240614_react@19.0.0-rc-fb9a90fa48-20240614_sass@1.77.6/node_modules/next/dist/server/base-server.js:1419:42)
    at async responseGenerator (/Users/gyozo/react/m15/node_modules/.pnpm/next@15.0.0-rc.0_react-dom@19.0.0-rc-fb9a90fa48-20240614_react@19.0.0-rc-fb9a90fa48-20240614_sass@1.77.6/node_modules/next/dist/server/base-server.js:1640:40)
    at async DevServer.renderToResponseWithComponentsImpl (/Users/gyozo/react/m15/node_modules/.pnpm/next@15.0.0-rc.0_react-dom@19.0.0-rc-fb9a90fa48-20240614_react@19.0.0-rc-fb9a90fa48-20240614_sass@1.77.6/node_modules/next/dist/server/base-server.js:1665:28)
    at async DevServer.renderPageComponent (/Users/gyozo/react/m15/node_modules/.pnpm/next@15.0.0-rc.0_react-dom@19.0.0-rc-fb9a90fa48-20240614_react@19.0.0-rc-fb9a90fa48-20240614_sass@1.77.6/node_modules/next/dist/server/base-server.js:1978:24)
    at async DevServer.renderToResponseImpl (/Users/gyozo/react/m15/node_modules/.pnpm/next@15.0.0-rc.0_react-dom@19.0.0-rc-fb9a90fa48-20240614_react@19.0.0-rc-fb9a90fa48-20240614_sass@1.77.6/node_modules/next/dist/server/base-server.js:2016:32)
    at async DevServer.pipeImpl (/Users/gyozo/react/m15/node_modules/.pnpm/next@15.0.0-rc.0_react-dom@19.0.0-rc-fb9a90fa48-20240614_react@19.0.0-rc-fb9a90fa48-20240614_sass@1.77.6/node_modules/next/dist/server/base-server.js:908:25)
    at async NextNodeServer.handleCatchallRenderRequest (/Users/gyozo/react/m15/node_modules/.pnpm/next@15.0.0-rc.0_react-dom@19.0.0-rc-fb9a90fa48-20240614_react@19.0.0-rc-fb9a90fa48-20240614_sass@1.77.6/node_modules/next/dist/server/next-server.js:273:17)
    at async DevServer.handleRequestImpl (/Users/gyozo/react/m15/node_modules/.pnpm/next@15.0.0-rc.0_react-dom@19.0.0-rc-fb9a90fa48-20240614_react@19.0.0-rc-fb9a90fa48-20240614_sass@1.77.6/node_modules/next/dist/server/base-server.js:804:17)
    at async /Users/gyozo/react/m15/node_modules/.pnpm/next@15.0.0-rc.0_react-dom@19.0.0-rc-fb9a90fa48-20240614_react@19.0.0-rc-fb9a90fa48-20240614_sass@1.77.6/node_modules/next/dist/server/dev/next-dev-server.js:339:20
    at async Span.traceAsyncFn (/Users/gyozo/react/m15/node_modules/.pnpm/next@15.0.0-rc.0_react-dom@19.0.0-rc-fb9a90fa48-20240614_react@19.0.0-rc-fb9a90fa48-20240614_sass@1.77.6/node_modules/next/dist/trace/trace.js:157:20)
    at async DevServer.handleRequest (/Users/gyozo/react/m15/node_modules/.pnpm/next@15.0.0-rc.0_react-dom@19.0.0-rc-fb9a90fa48-20240614_react@19.0.0-rc-fb9a90fa48-20240614_sass@1.77.6/node_modules/next/dist/server/dev/next-dev-server.js:336:24)
    at async invokeRender (/Users/gyozo/react/m15/node_modules/.pnpm/next@15.0.0-rc.0_react-dom@19.0.0-rc-fb9a90fa48-20240614_react@19.0.0-rc-fb9a90fa48-20240614_sass@1.77.6/node_modules/next/dist/server/lib/router-server.js:175:21)
    at async handleRequest (/Users/gyozo/react/m15/node_modules/.pnpm/next@15.0.0-rc.0_react-dom@19.0.0-rc-fb9a90fa48-20240614_react@19.0.0-rc-fb9a90fa48-20240614_sass@1.77.6/node_modules/next/dist/server/lib/router-server.js:354:24)
    at async requestHandlerImpl (/Users/gyozo/react/m15/node_modules/.pnpm/next@15.0.0-rc.0_react-dom@19.0.0-rc-fb9a90fa48-20240614_react@19.0.0-rc-fb9a90fa48-20240614_sass@1.77.6/node_modules/next/dist/server/lib/router-server.js:378:13)
    at async Server.requestListener (/Users/gyozo/react/m15/node_modules/.pnpm/next@15.0.0-rc.0_react-dom@19.0.0-rc-fb9a90fa48-20240614_react@19.0.0-rc-fb9a90fa48-20240614_sass@1.77.6/node_modules/next/dist/server/lib/start-server.js:142:13)
 PATCH /api/pages/8?draft=true&autosave=true&locale=en 500 in 36ms
```

### Reproduction Steps

```
UploadFeature({
  collections: {
    media: {
      fields: [
        {
          name: 'alt',
          label: 'Alt Text',
          type: 'text',
        },
        { ... },
      ],
    },
  },
})
```

### Adapters and Plugins

richtext-lexical, UploadFeature

**Comment by vixalien** [comment]
having the same error, except for me the application doesn't start all. Here's my code.

```js
UploadFeature({
  collections: {
    media: {
      fields: [
        {
          name: "caption",
          label: "Caption",
          type: "richText",
          editor: lexicalEditor({
            features: () => [
              ParagraphFeature(),
              ...defaultCMSFeatures,
            ],
          }),
        },
      ],
    },
  },
}),
```

Stack:

```
file:///home/alien/sites/rwarri.org/node_modules/.pnpm/payload@3.0.0-beta.53_@swc+core@1.6.3_@swc+helpers@0.5.11__@swc+types@0.1.8_graphql@16.8.2_typescript@5.5.2/node_modules/payload/dist/fields/hooks/afterRead/promise.js:23
    const shouldHoistLocalizedValue = flattenLocales && fieldAffectsData(field) && typeof siblingDoc[field.name] === 'object' && siblingDoc[field.name] !== null && field.localized && locale !== 'all' && req.payload.config.localization;
                                                                                                    ^

TypeError: Cannot read properties of null (reading 'caption')
    at promise (file:///home/alien/sites/rwarri.org/node_modules/.pnpm/payload@3.0.0-beta.53_@swc+core@1.6.3_@swc+helpers@0.5.11__@swc+types@0.1.8_graphql@16.8.2_typescript@5.5.2/node_modules/payload/dist/fields/hooks/afterRead/promise.js:23:101)
    at file:///home/alien/sites/rwarri.org/node_modules/.pnpm/payload@3.0.0-beta.53_@swc+core@1.6.3_@swc+helpers@0.5.11__@swc+types@0.1.8_graphql@16.8.2_typescript@5.5.2/node_modules/payload/dist/fields/hooks/afterRead/traverseFields.js:4:28
    at Array.forEach (<anonymous>)
    at Module.traverseFields (file:///home/alien/sites/rwarri.org/node_modules/.pnpm/payload@3.0.0-beta.53_@swc+core@1.6.3_@swc+helpers@0.5.11__@swc+types@0.1.8_graphql@16.8.2_typescript@5.5.2/node_modules/payload/dist/fields/hooks/afterRead/traverseFields.js:3:12)
    at hooks.afterRead (/home/alien/sites/rwarri.org/node_modules/.pnpm/@payloadcms+richtext-lexical@3.0.0-beta.53_oya34g7zejj37344dtca3uxyqm/node_modules/@payloadcms/richtext-lexical/dist/index.js:286:59)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async file:///home/alien/sites/rwarri.org/node_modules/.pnpm/payload@3.0.0-beta.53_@swc+core@1.6.3_@swc+helpers@0.5.11__@swc+types@0.1.8_graphql@16.8.2_typescript@5.5.2/node_modules/payload/dist/fields/hooks/afterRead/promise.js:515:49
    at async promise (file:///home/alien/sites/rwarri.org/node_modules/.pnpm/payload@3.0.0-beta.53_@swc+core@1.6.3_@swc+helpers@0.5.11__@swc+types@0.1.8_graphql@16.8.2_typescript@5.5.2/node_modules/payload/dist/fields/hooks/afterRead/promise.js:477:21)
    at async Promise.all (index 6)
    at async afterRead (file:///home/alien/sites/rwarri.org/node_modules/.pnpm/payload@3.0.0-beta.53_@swc+core@1.6.3_@swc+helpers@0.5.11__@swc+types@0.1.8_graphql@16.8.2_typescript@5.5.2/node_modules/payload/dist/fields/hooks/afterRead/index.js:41:5)

Node.js v22.1.0
```

**Comment by AlessioGr** [comment]
Fixed in [`7c8857a` (#6885)](https://github.com/payloadcms/payload/pull/6885/commits/7c8857ab12a7965f170737a2ae28e2c4bfaac76c).

Please re-open if this issue still appears in the next 3.0.0-beta.54 release

**Comment by carlonoelle** [comment]
I have a related problem i think, my extra fields aren't showing up at all:


https://github.com/user-attachments/assets/3cc9b6c6-8b52-468a-b102-f4d9df81c7ae



My RichText Field Config:
```
import { type RichTextField } from "payload"
import { FeatureProviderServer, UploadFeature, lexicalEditor } from "@payloadcms/richtext-lexical"
import link from "../link"
import { defaultFeatures } from "./defaultFeatures"
import deepMerge from "@/payload/utilities/deepMerge"

type RichText = (
  overrides?: Partial<RichTextField>,
  additions?: {
    features?: FeatureProviderServer[]
  }
) => RichTextField

const richText: RichText = (
  overrides,
  additions = {
    features: [],
  }
) =>
  deepMerge<RichTextField, Partial<RichTextField>>(
    {
      name: "richText",
      editor: lexicalEditor({
        features: () => [
          ...[...defaultFeatures, ...(additions.features || [])],
          UploadFeature({
            collections: {
              media: {
                hasExtraFields: true,
                fields: [
                  {
                    name: "caption",
                    editor: lexicalEditor({
                      features: () => [...defaultFeatures],
                    }),
                    label: "Caption",
                    type: "richText",
                  },
                  {
                    name: "alignment",
                    label: "Alignment",
                    options: [
                      {
                        label: "Left",
                        value: "left",
                      },
                      {
                        label: "Center",
                        value: "center",
                      },
                      {
                        label: "Right",
                        value: "right",
                      },
                    ],
                    type: "radio",
                  },
                  {
                    name: "enableLink",
                    label: "Enable Link",
                    type: "checkbox",
                  },
                  /* link({
                    appearances: false,
                    disableLabel: true,
                    overrides: {
                      admin: {
                        condition: (_: any, data: any) => Boolean(data?.enableLink),
                      },
                    },
                  }), */
                ],
              },
            },
          }),
        ],
      }),
      required: true,
      type: "richText",
    },
    overrides || {}
  )

export default richText
```

I am using all payload packages at Version 3.0.0-beta.60.
